yang/frr-bgp-common-structure.yang

```diff
@@ -0,0 +1,807 @@
+    prefix frr-interface;
+  }
+
+  import ietf-bgp-types {
+    prefix bt;
+  }
+
+  import frr-bgp-types {
+    prefix frr-bt;
+  }
+
+  include frr-bgp-common;
+
+  organization
+    "FRRouting";
+  contact
+    "FRR Users List: <mailto:frog@lists.frrouting.org> FRR Development
+     List: <mailto:dev@lists.frrouting.org>";
+  description
+    "This submodule contains general data definitions for use in BGP.
+
+     Copyright 2020 FRRouting
+
+     Redistribution and use in source and binary forms, with or without
+     modification, are permitted provided that the following conditions
+     are met:
+
+     1. Redistributions of source code must retain the above copyright notice,
+     this list of conditions and the following disclaimer.
+
+     2. Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+
+     THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+     \"AS IS\" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+     LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+     A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+     HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+     SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+     LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+     DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+     THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+     (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+     OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.";
+
+  revision 2019-12-03 {
+    description
+      "Initial revision.";
+  }
+
+  grouping structure-neighbor-group-ebgp-multihop {
+    description
+      "Structural grouping used to include EBGP multi-hop
+       configuration for both BGP neighbors and peer groups.";
+    container ebgp-multihop {
+      description
+        "EBGP multi-hop parameters for the BGP group.";
+      choice hop-count {
+        case default-hop-count {
+          leaf enabled {
+            type boolean;
+            default "false";
+            description
+              "When enabled the referenced group or neighbors are
+               permitted to be indirectly connected - including cases
+               where the TTL can be decremented between the BGP peers.";
+          }
+        }
+
+        case max-hop-count {
+          leaf multihop-ttl {
+            type uint8 {
+              range "1..255";
+            }
+            description
+              "Time-to-live value to use when packets are sent to the
+               referenced group or neighbors and ebgp-multihop is
+               enabled.";
+          }
+        }
+      }
+
+      leaf disable-connected-check {
```

Comment:
Does this override the equivalent global setting?

Reply:
This is getting mapped to
#define PEER_FLAG_DISABLE_CONNECTED_CHECK   (1 << 6) /* disable-connected-check */